sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -587,13 +587,24 @@ fromClause
     ;
 
 aggregationClause
-    : GROUP BY groupingExpressions+=expression (',' groupingExpressions+=expression)* (
+    : GROUP BY groupingExpressionWithGroupingAnalytics+=groupByClause
+      (',' groupingExpressionWithGroupingAnalytics+=groupByClause)*
+    | GROUP BY groupingExpressions+=expression (',' groupingExpressions+=expression)* (
       WITH kind=ROLLUP
     | WITH kind=CUBE
     | kind=GROUPING SETS '(' groupingSet (',' groupingSet)* ')')?
     | GROUP BY kind=GROUPING SETS '(' groupingSet (',' groupingSet)* ')'
     ;
 
+groupByClause
+    : groupingAnalytics
+    | expression
+    ;
+
+groupingAnalytics
+    : (ROLLUP | CUBE | GROUPING SETS)  '(' groupingSet (',' groupingSet)* ')'
```

Comment:
If this PR includes the `groupingSet` support for `CUBE`/`ROLLUP`, please add tests for the new feature.

Reply:
`(ROLLUP | CUBE | GROUPING SETS) ` => `kind=(ROLLUP | CUBE | GROUPING SETS) `?

Reply:
> `(ROLLUP | CUBE | GROUPING SETS) ` => `kind=(ROLLUP | CUBE | GROUPING SETS) `?

Kind can't be assigned to a set.

Reply:
> If this PR includes the `groupingSet` support for `CUBE`/`ROLLUP`, please add tests for the new feature.

yea

Reply:
> Kind can't be assigned to a set.

https://github.com/apache/spark/blob/d082ad0abfe0bc26760626ae0ecb415a8d508a1f/sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4#L776 ?

Reply:
![image](https://user-images.githubusercontent.com/46485123/100415463-29996800-30b7-11eb-80ea-830f678491d7.png)
![image](https://user-images.githubusercontent.com/46485123/100415472-2d2cef00-30b7-11eb-8a84-b166f6a59760.png)
==

Reply:
Sorry for my mistake in https://github.com/apache/spark/pull/30212#discussion_r529502777

`Kind can't be assigned to a set. ` is wrong , should be `Kind can't be assigned to not a set.`

Reply:
Ah, ok.